osmaxx-py/osmaxx/excerptexport/static/excerptexport/scripts/formValidation.js

```diff
@@ -32,15 +66,60 @@
                     checkbox.setCustomValidity('Please choose minimal one export format!');
                 }
             }.bind(this));
+            this.setSubmitButtonState();
         }.bind(this);
 
+        this.validateExcerptBounds = function() {
+            var allowedMaxSize = 250*1024*1024;
+
+            var north = document.getElementById('id_north').value;
+            var east = document.getElementById('id_east').value;
+            var south = document.getElementById('id_south').value;
+            var west = document.getElementById('id_west').value;
+
+            jQuery.getJSON(
+                '/api/estimated_file_size/',
+                { 'north': north, 'east': east, 'south': south, 'west': west },
+                function(data) {
+                    var estimatedFileSize = Number(data['estimated_file_size_in_bytes']);
+                    this.validity['excerptBounds'] = estimatedFileSize < allowedMaxSize;
+                    this.setSubmitButtonState();
+                    
+                    this.excerptBoundInputFields.forEach(function(excerptBoundInputField) {
+                        if(this.validity['excerptBounds']) {
+                            excerptBoundInputField.setCustomValidity('');
+                            document.getElementById('excerpt-validation').textContent = '';
+                        } else {
+                            var howMuchTooLarge = estimatedFileSize ? Math.round(100 / allowedMaxSize * estimatedFileSize - 100) + '% ': '';
```

Comment:
Would `Math.ceil(...)` be better suited here? I don't think we want confusing `Excerpt 0% too large!` messages.